File: app/src/main/java/com/example/maunorafiq/pawangcuaca/presentation/mapper/ForecastModelDataMapper.java
package com.example.maunorafiq.pawangcuaca.presentation.mapper;

import android.util.Log;

import com.example.maunorafiq.pawangcuaca.presentation.internal.di.PerActivity;
import com.example.maunorafiq.pawangcuaca.presentation.model.ForecastModel;
import com.icehousecorp.maunorafiq.domain.forecast.Forecast;

import javax.inject.Inject;

/**
 * Created by maunorafiq on 11/29/16.
 */

@PerActivity
public class ForecastModelDataMapper {

    private final String TAG = this.getClass().getSimpleName();

    private final WeatherModelDataMapper weatherModelDataMapper;

    @Inject
    public ForecastModelDataMapper(WeatherModelDataMapper weatherModelDataMapper) {
        this.weatherModelDataMapper = weatherModelDataMapper;
    }

    public ForecastModel transform(Forecast forecast) {
        ForecastModel forecastModel = new ForecastModel(forecast.getCityName());
        forecastModel.setForecastList(weatherModelDataMapper.transform(forecast.getForecastList()));
        return forecastModel;
    }
}

File: app/src/main/java/com/example/maunorafiq/pawangcuaca/presentation/mapper/ForecastModelDataMapper.kt
package com.example.maunorafiq.pawangcuaca.presentation.mapper

import android.util.Log

import com.example.maunorafiq.pawangcuaca.presentation.internal.di.PerActivity
import com.example.maunorafiq.pawangcuaca.presentation.model.ForecastModel
import com.icehousecorp.maunorafiq.domain.forecast.Forecast

import javax.inject.Inject

/**
 * Created by maunorafiq on 11/29/16.
 */

@PerActivity
class ForecastModelDataMapper
@Inject
constructor(private val weatherModelDataMapper: WeatherModelDataMapper) {

    private val TAG = this.javaClass.simpleName

    fun transform(forecast: Forecast): ForecastModel {
        val forecastModel = ForecastModel(forecast.cityName)
        forecastModel.forecastList = weatherModelDataMapper.transform(forecast.forecastList)
        return forecastModel
    }
}
